### DISCUSSION TITLE
Use selenium-wire with Pylenium

### DISCUSSION THREAD

**Discussion by szabolcslevay** [ElSnoMan/pyleniumio#239] [Q&A] [answered]
Is there a way to use [selenium-wire](https://github.com/wkeeling/selenium-wire) with Pylenium?
It would be great to have a solution to access the underlying requests made by the browser.

Thank you,

**Comment by ElSnoMan** [comment]
This looks really cool! Selenium 4 introduced the [BiDirectional API (bidi)](https://www.selenium.dev/documentation/webdriver/bidirectional/bidi_api/) and the [Chrome DevTools Protocol (CDP)](https://www.selenium.dev/documentation/webdriver/bidirectional/chrome_devtools/) that may have some functionality we can use, but the Python docs aren't the best.

I do like what Selenium Wire is doing, but we have to think about how we introduce it without making too many breaking changes to existing Pylenium users. I'll be exploring this some more, but you're more than welcome to give it a try and pose some solutions or even submit a Pull Request!

**Reply by szabolcs-levay** [reply]
Thank you for looking into this, @ElSnoMan!

**Reply by ElSnoMan** [reply]
@szabolcslevay How do you get selenium-wire to work on Selenium Grid? I'm using RemoteDriver but I'm getting all sorts of crazy errors and can't seem to fix it. I have Chrome and Firefox working with selenium-wire, but now Remote (which used to be trivial) is adding a lot of complexity 😕

**Reply by szabolcs-levay** [reply]
Unfortunately, I didn’t try because it wasn’t among our use cases. We don't rely on Selenium Grid for parallelization.

**Reply by ElSnoMan** [reply]
Good to know! Pylenium 1.15.1 has selenium-wire for Chrome and Firefox, but I left out RemoteDriver for now. Hopefully that also works for your solution. What are you using instead of Grid?

**Reply by szabolcs-levay** [reply]
> Pylenium 1.15.1 has selenium-wire for Chrome and Firefox

That's great! Thank you for taking the time to do this!

> What are you using instead of Grid?

We are using a custom Docker-based solution everything combined in a Jenkinsfile in a scripted pipeline format.

**Reply by ElSnoMan** [reply]
Very cool! If you're willing, I'd love to hear more about this in detail. Maybe in DM if you don't want it to be public or discussed here 😁

**Comment by ElSnoMan** [comment, marked as answer]
As of Pylenium `v1.15.1`, [selenium-wire](https://github.com/wkeeling/selenium-wire) is now integrated! 🎉 

Local Chrome and Firefox browsers can do all of the selenium-wire things, so check out their documentation 😄

**Comment by serdnago** [comment]
Aloha!,
Do you have an example of pyleniumio integration with Selenium -wire?
Mahalo!

**Reply by serdnago** [reply]
Actually I figure it out, very simple, thank you!!!

**Comment by ElSnoMan** [comment]
https://docs.pylenium.io/guides/inspect-network-requests

This is one example, but since you have access to the WebDriver instance,
you should be able to do what you need.

Is there a specific example you're looking for?
